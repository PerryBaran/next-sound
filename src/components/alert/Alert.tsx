import css from "./alert.module.css"

export default function Alert({ message }: { message: string }) {
  if (!message) return null

  return <p className={css["alert"]}>{message}</p>
}
