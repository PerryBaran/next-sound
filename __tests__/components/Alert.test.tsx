import { render, screen } from "@testing-library/react"
import Alert from "../../src/components/alert/Alert"

describe("Alert", () => {
  test("if passed a truthy message, message is rendered to screen", () => {
    const message = "test"
    render(<Alert message={message} />)

    expect(screen.getByText(message)).toBeInTheDocument()
  })

  test("if passed a falsey message, returns null", () => {
    const { container } = render(<Alert message="" />)

    expect(container).toBeEmptyDOMElement()
  })

  describe("snapshots", () => {
    test("truthy message snapshot", () => {
      const { asFragment } = render(<Alert message="test" />)

      expect(asFragment()).toMatchSnapshot()
    })

    test("falsey message snapshot", () => {
      const { asFragment } = render(<Alert message="" />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
