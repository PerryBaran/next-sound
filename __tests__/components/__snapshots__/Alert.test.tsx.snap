// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert snapshots falsey message snapshot 1`] = `<DocumentFragment />`;

exports[`Alert snapshots truthy message snapshot 1`] = `
<DocumentFragment>
  <p
    class="alert"
  >
    test
  </p>
</DocumentFragment>
`;
